finality-aleph/src/new_network/service.rs

```diff
@@ -204,52 +226,95 @@ mod tests {
+        pub network: MockNetwork,
+        pub mock_io: MockIO,
+        // `TaskManager` can't be dropped for `SpawnTaskHandle` to work
+        task_manager: TaskManager,
+    }
+
+    impl TestData {
+        async fn prepare() -> Self {
+            let task_executor: TaskExecutor =
+                (move |future, _task_type| tokio::spawn(future).map(|_| ())).into();
+            let task_manager = TaskManager::new(task_executor, None).unwrap();
+            // Prepare communication with service
+            let (mock_messages_for_user, messages_from_user) = mpsc::unbounded();
+            let (messages_for_user, mock_messages_from_user) = mpsc::unbounded();
+            let (mock_commands_for_manager, commands_from_manager) = mpsc::unbounded();
+            let io = IO {
+                messages_from_user,
+                messages_for_user,
+                commands_from_manager,
+            };
+            let mock_io = MockIO {
+                messages_for_user: mock_messages_for_user,
+                messages_from_user: mock_messages_from_user,
+                commands_for_manager: mock_commands_for_manager,
+            };
+            // Prepare service
+            let (event_stream_oneshot_tx, event_stream_oneshot_rx) = oneshot::channel();
+            let network = MockNetwork::new(event_stream_oneshot_tx);
+            let service = Service::new(network.clone(), task_manager.spawn_handle(), io);
+            let (exit_tx, exit_rx) = oneshot::channel();
+            let task_handle = async move {
+                tokio::select! {
+                    _ = service.run() => { },
+                    _ = exit_rx => { },
+                };
             };
-        };
-        let service_handle = tokio::spawn(task_handle);
+            let service_handle = tokio::spawn(task_handle);
+            // wait till service takes the event_stream
+            event_stream_oneshot_rx.await.unwrap();
+
+            // `TaskManager` needs to be passed.
+            Self {
+                service_handle,
+                exit_tx,
+                network,
+                mock_io,
+                task_manager,
+            }
+        }
 
-        // wait till service takes the event_stream
-        event_stream_oneshot_rx.await.unwrap();
+        async fn cleanup(self) {
+            self.exit_tx.send(()).ok();
+            self.service_handle.await.unwrap();
+            self.network.close_channels();
+        }
 
-        (service_handle, exit_tx, network, mock_io)
+        // We do this only to make sure that NotificationStreamOpened/NotificationStreamClosed events are handled
+        async fn wait_for_events_handled(&mut self) {
+            let identity = MockNetworkIdentity::new().identity();
+            self.network.emit_event(Event::SyncConnected {
+                remote: identity.1.into(),
+            });
+            let expected = ScMultiaddr::empty().with(ScProtocol::P2p(identity.1 .0.into()));
```

Comment:
`.1 .0.` has an unexpected space and is a bit, uh, hard to read. Don't worry about the latter if it's hard to fix, but please remove the space.

Reply:
Unfortunately this is the way rust formatter wants it to be :( I had a comment about that in the previous PR too